great_expectations/dataset/pandas_dataset.py

```diff
@@ -1577,6 +1634,76 @@ def estimated_cdf(x):
+        self,
+        column_A,
+        column_B,
+        bins_A=None,
+        bins_B=None,
+        n_bins_A=None,
+        n_bins_B=None,
+        threshold=0.05,
+        ignore_missings=None,
+        result_format=None,
+        include_config=True,
+        catch_exceptions=None,
+        meta=None,
+    ):
+        """
+        Expect the values in column_A to be independent of those in column_B.
+
+        Args:
+            column_A (str): The first column name
+            column_B (str): The second column name
+            threshold (float): Maximum allowed value of cramers V for expectation to pass.
+
+        Keyword Args:
+            bins_A (list of float): Bins for column_A if numeric.
+            bins_B (list of float): Bins for column_B if numeric.
+            n_bins_A (int): Number of bins for column_A if numeric. Ignored if bins_A is not None.
+            n_bins_B (int): Number of bins for column_B if numeric. Ignored if bins_B is not None.
+            ignore_missings (bool): \
+                If True, ignore columns where either column has missing. If False, treat missing values as
+                another category (only for methods ...).
+
+        Other Parameters:
+            result_format (str or None): \
+                Which output mode to use: `BOOLEAN_ONLY`, `BASIC`, `COMPLETE`, or `SUMMARY`.
+                For more detail, see :ref:`result_format <result_format>`.
+            include_config (boolean): \
+                If True, then include the expectation config as part of the result object. \
+                For more detail, see :ref:`include_config`.
+            catch_exceptions (boolean or None): \
+                If True, then catch exceptions and include them as part of the result object. \
+                For more detail, see :ref:`catch_exceptions`.
+            meta (dict or None): \
+                A JSON-serializable dictionary (nesting allowed) that will be included in the output without \
+                modification. For more detail, see :ref:`meta`.
+
+        Returns:
+            A JSON-serializable expectation result object.
+
+            Exact fields vary depending on the values passed to :ref:`result_format <result_format>` and
+            :ref:`include_config`, :ref:`catch_exceptions`, and :ref:`meta`.
+
+        """
+        crosstab = self.get_crosstab(
+            column_A, column_B, bins_A, bins_B, n_bins_A, n_bins_B
+        )
+        chi2_result = stats.chi2_contingency(crosstab)
+        # See e.g. https://en.wikipedia.org/wiki/Cram%C3%A9r%27s_V
+        cramers_V = np.sqrt(chi2_result[0] / self.get_row_count() / min(crosstab.shape))
+        return_obj = {
+            "success": cramers_V <= threshold,
+            "result": {
+                "observed_value": cramers_V,
+                "unexpected_list": crosstab,
+                "details": {"crosstab": crosstab},
+            },
+        }
+        return return_obj
```

Comment:
Could we move this implementation into the parent `Dataset` class, since the children will implement the relevant `get_crosstabs` method?

Reply:
IIUC this would require implementing `get_crosstabs`. See [here](https://github.com/great-expectations/great_expectations/pull/1608/files#diff-8109b0b2ad0be8c0042c91bd24c68953R4161).

Reply:
Yes, you're right in principle, but I think it's appropriate in this PR to still do that move and then add unimplemented `get_crosstab` methods for both SqlAlchemy and Spark. 

If you'd like to combine to a more fully cross-backend implementation, I see a good path to that: for spark, we could get an implementation almost immediately using `bucketizer` analogously to how you're using `digitize`. 

For SqlAlchemy, we could dynamically construct a case statement from buckets in similar fashion. I'd be happy to add those implementations in a separate PR.

Reply:
...correction; the above should apply to *get_binned_values*

get_crosstabs is a similar story for spark using the crosstab option, but I'm not sure there's a good general SQL option. Still, we could leave it unimplemented for some backends to move forward.

Reply:
I moved the expectation to `Dataset` and added a `Dataset.get_crosstab` that raises `NotImplementedError`. Since `get_crosstab` does aggregation of column data in pandas, the actual function should stay part of `PandasDataset` (and not `Dataset`) IMO.